daemon/algod/api/algod.oas2.json

```diff
@@ -1433,6 +1433,145 @@
         }
       ]
     },
+    "/v2/applications/{application-id}/boxes": {
+      "get": {
+        "description": "Given an application ID, it returns the box names of that application. No particular ordering is guaranteed.",
+        "produces": [
+          "application/json"
+        ],
+        "schemes": [
+          "http"
+        ],
+        "summary": "Get box names for a given application.",
+        "operationId": "GetApplicationBoxes",
+        "parameters": [
+          {
+            "type": "integer",
+            "description": "An application identifier",
+            "name": "application-id",
+            "in": "path",
+            "required": true
+          },
+          {
+            "type": "integer",
+            "description": "Max number of box names to return. If max is not set, or max == 0, returns all box-names.",
+            "name": "max",
+            "in": "query"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "$ref": "#/responses/BoxesResponse"
+          },
+          "400": {
+            "description": "Bad Request",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          },
+          "401": {
+            "description": "Invalid API Token",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          },
+          "500": {
+            "description": "Internal Error",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          },
+          "default": {
+            "description": "Unknown Error"
+          }
+        }
+      },
+      "parameters": [
+        {
+          "type": "integer",
+          "name": "application-id",
+          "in": "path",
+          "required": true
+        }
+      ]
+    },
+    "/v2/applications/{application-id}/box": {
```

Comment:
Having /v2/applications/{id}/boxes and /v2/applications/{id}/box is totally out of place in our REST API and these sorts of APIs in general. When identifying a resource by name it should be specified directly in the required path parameter.

Specifically, this should be /v2/applications/{id}/boxes/{resource}

Reply:
That was the original approach, but there isn't a standard way to encode binary data in a url.  The {resource} here can have slashes and non-utf8 bytes in it, for example.  So you could b64 encode it? But contrary to everywhere else we b64 encode, here it would have to be url-safe, not std encoding.  It seemed like a mess to wedge the binary name of a box into a url segment - and it's not as though we have a precedent of b64 encoding binary data in our REST urls.  We could accept the various encodings that the goal cli accepts (which are ascii), but that would also violate REST expectations because there would be many synonyms for any particular resource. Futhermore, *they* can have slashes in them.

Reply:
Thanks for the explanation, I was looking for the technical reasoning behind the change but didn't find much. One thing I'm still not clear on, is whether query parameters solve the problem you described. The binary data is still going to be in the URL, doesn't it still need to be url encoded?

There is some precedent for binary data in a URL in the "b32 without padding" format that we use for addresses. I believe this problem is one of the reasons we have that format for addresses. Given how you imagine people using boxes, do you think it's worth having a safe format like this for box addresses?

Reply:
I think query parameters fix the problem because we introduced (really just ported over) an encoding scheme for the binary data.  That is, just as `goal --app-arg` accepts things like `str:hello`, `int:76`, and `addr:GAFSAAA...`, that is the form expected in the query parameter. So there's no longer a need to escape anything.  We probably could have done that in the path, except for the feeling that doing so feels wrong, since there is no canonical name for the resource.

If you *do* need to escape something, for example you'd like to look up `str:pb&j`, you can escape the & to avoid screwing up the query parameter parsing.  But in a path, you can't. If you escape a "/" as %57, it will get turned into a "/" early enough to foul up how the URL matching is performed.  It would look like two path segments.

And I think that encoding scheme, inherited from `goal` probably addresses the need you mentioned.  There is a pretty nice way to write out boxes. Since their form is not always the same, we can't select the right one on output (so the return values of endpoints always us b64:GAFGS...), but it allows nicely writing them down for input.

Reply:
(I could be wrong about that pb&j thing, btw.  It may be that you need to select an encoding that doesn't have a & in it.)

Reply:
After a verbal with @algoanne and @winder, we landed on leaving the PR as is.  Resolving + sharing more context below.

* From a consistency perspective, it'd be preferable to avoid the deviation.  Though the tradeoff between time-to-market and user inconvenience balances slightly towards keeping as is.
* I take it as an action item to speak up about API inconsistencies earlier in the development process.